Services/IotHub/IotHubConnectionStringManager.cs

```diff
@@ -131,19 +141,28 @@ public void ValidateConnectionString(string connectionString)
                 throw new InvalidIotHubConnectionStringFormatException(message);
             }
 
+            // check if KeyName permissions are either "iothubowner" or "service"
+            if (!(match.Groups[CONNSTRING_REGEX_KEYNAME].Value == OWNER_KEYNAME ||
+                  match.Groups[CONNSTRING_REGEX_KEYNAME].Value == SERVICE_KEYNAME))
```

Comment:
as discussed offline, we cannot use this kind of validation, because users can create custom connection strings with custom keynames